app/controllers/public/cfs/new-session.js

```diff
@@ -1,20 +1,17 @@
 import Controller from '@ember/controller';
-
 export default Controller.extend({
   actions: {
-    save() {
-      this.set('isLoading', true);
-      this.get('model.session').save()
-        .then(() => {
-          this.get('notify').success(this.get('l10n').t('Session details have been saved'));
-          this.transitionToRoute('public.cfs.index');
-        })
-        .catch(() => {
-          this.get('notify').error(this.get('l10n').t('Oops something went wrong. Please try again'));
-        })
-        .finally(() => {
-          this.set('isLoading', false);
-        });
+    async save(speakerDetails) {
+      try {
+        this.set('isLoading', true);
+        await this.get('model.session').save();
+        speakerDetails.sessions.pushObject(this.get('model.session'));
+        await this.get('model.session').save();
```

Comment:
why is this twice?

Reply:
@CosmicCoder96 once it stores the details of the form and then the speaker is pushed to its relationship and then saved again. If written once the speaker doesn't get added to the session.